src/CurrencyNetwork.ts

```diff
@@ -56,33 +66,45 @@ export class CurrencyNetwork {
     userAddress: string
   ): Promise<UserOverview> {
     await this._checkAddresses([networkAddress, userAddress])
-    const [ overview, decimals ] = await Promise.all([
+    const [
+      overview,
+      { networkDecimals }
+    ] = await Promise.all([
       this._utils.fetchUrl<UserOverviewRaw>(`networks/${networkAddress}/users/${userAddress}`),
       this.getDecimals(networkAddress)
     ])
     return {
-      balance: this._utils.formatToAmount(overview.balance, decimals),
-      given: this._utils.formatToAmount(overview.given, decimals),
-      received: this._utils.formatToAmount(overview.received, decimals),
-      leftGiven: this._utils.formatToAmount(overview.leftGiven, decimals),
-      leftReceived: this._utils.formatToAmount(overview.leftReceived, decimals)
+      balance: this._utils.formatToAmount(overview.balance, networkDecimals),
+      given: this._utils.formatToAmount(overview.given, networkDecimals),
+      received: this._utils.formatToAmount(overview.received, networkDecimals),
+      leftGiven: this._utils.formatToAmount(overview.leftGiven, networkDecimals),
+      leftReceived: this._utils.formatToAmount(overview.leftReceived, networkDecimals)
     }
   }
 
   /**
-   * Returns the decimals specified in a currency network.
+   * Returns the network decimals and interest decimals specified in a currency network.
    * @param networkAddress Address of currency network.
    * @param decimals If decimals are known they can be provided manually.
    */
-  public async getDecimals (networkAddress: string, decimals?: number): Promise<number> {
+  public async getDecimals (
```

Comment:
I think that this function is weird. That it will if you give it some decimals will just return them. This also confused Thore during the hackathon. I think this function should just return the decimals or none if it cant find one, and the caller will need to decide whether to use this function or nor. 
I will open a refactor issue if you agree.

Reply:
Yes please do so. The function always felt a bit hacky also in my eyes...

Reply:
Opened and issue